app/models/package.scala

```diff
@@ -5,7 +5,7 @@ package object models {
   import play.api.Logger
   import play.api.Play.current
 
-  lazy val db = {
+  def db = {
```

Comment:
Alpha, I added back your change so it is a method and not a lazy val with a
TODO message.

On Mon, Jan 12, 2015 at 9:44 AM, Bryan Castillo <castillo.bryan@gmail.com>
wrote:

> Interesting.  I think I understand why that is happening now.  I thought
> about it more and your change to a regular method (def) instead of a lazy
> val isn't as inefficient as I thought.  I believe a slick Database object
> will be created every time, but the play application should cache the
> DataSource itself.  I think we can change it back to what you had.
>
> This gets me thinking that using DI would make this easier.  The Database
> object should only be created on Application startup, in the Global object
> and it should be destroyed during application shutdown.
>
>
>
> On Mon, Jan 12, 2015 at 8:41 AM, Alpha Chen <notifications@github.com>
> wrote:
>
>> In app/models/package.scala
>> <https://github.com/fkearney8/151Challenge/pull/6#discussion-diff-22802204>
>> :
>>
>> > @@ -5,7 +5,7 @@ package object models {
>> >    import play.api.Logger
>> >    import play.api.Play.current
>> >
>> > -  lazy val db = {
>> > +  def db = {
>>
>> Yeah, this change was required on my machine to get the tests to run.
>> Otherwise, I get this error:
>>
>> [error] Caused by java.sql.SQLException: Attempting to obtain a connection from a pool that has already been shutdown.
>>
>> I'll take a look at getting the tests to run against either an in-memory
>> db or a different postgres db.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/fkearney8/151Challenge/pull/6/files#r22802204>.
>>
>
>